source/test/TestApp/PlayFabTestAlloc.cpp

```diff
@@ -67,10 +68,28 @@ namespace PlayFabUnit
         testContext.Pass();
     }
 
+    void PlayFabTestAlloc::TestApiInstanceAlloc(TestContext& testContext)
+    {
+        // If anything fails here, it'll be a straight program crash
+
+        // Context
+        PlayFab::PlayFabAuthenticationInstanceAPI* testPtrApi = new PlayFab::PlayFabAuthenticationInstanceAPI();
+        delete testPtrApi;
+
+        std::shared_ptr<PlayFab::PlayFabAuthenticationInstanceAPI> testSpApi = std::make_shared<PlayFab::PlayFabAuthenticationInstanceAPI>();
+        testSpApi.reset();
+
+        std::shared_ptr<PlayFab::PlayFabAuthenticationInstanceAPI> testNSpApi = std::shared_ptr<PlayFab::PlayFabAuthenticationInstanceAPI>(new PlayFab::PlayFabAuthenticationInstanceAPI());
+        testNSpApi.reset();
+
+        testContext.Pass();
+    }
+
     void PlayFabTestAlloc::AddTests()
```

Comment:
This is a strange test.  What is the motivation behind testing allocation w/wo shared pointer etc?  This seems more like a test of the functionality of shared_ptr than it is a test of allocating a PlayFabAuthenticationInstanceIApi...

Reply:
The motivation is that the objects in this file have been known to seg-fault when being destroyed.
This was the big 3 hour full-team discussion about how the exe and the lib think the object is a different size.